bootstore/src/schemes/v0/fsm.rs

```diff
@@ -0,0 +1,820 @@
+    ///
+    /// This must be called after any API callback
+    pub fn drain_envelopes(&mut self) -> impl Iterator<Item = Envelope> + '_ {
+        self.envelopes.drain(..).chain(self.request_manager.drain_elements())
+    }
+
+    /// Put a request into an envelope and add it to `self.envelopes`
+    pub fn push_response(
+        &mut self,
+        to: Baseboard,
+        request_id: Uuid,
+        type_: ResponseType,
+    ) {
+        self.envelopes
+            .push(Envelope { to, msg: Response { request_id, type_ }.into() });
+    }
+
+    pub fn state(&self) -> &State {
+        &self.state
+    }
+
+    pub fn rack_init_failed(&self) -> bool {
+        self.rack_init_error.is_some()
+    }
+
+    /// This call is triggered locally on a single sled as a result of RSS
+    /// running. It may only be called once, which is enforced by checking to
+    /// see if we already are in `State::Uninitialized`.
+    ///
+    /// Persistence is required after a successful call to `init_rack`.
+    pub fn init_rack(
+        &mut self,
+        now: Instant,
+        rack_uuid: Uuid,
+        initial_membership: BTreeSet<Baseboard>,
+    ) -> Result<(), ApiError> {
+        if let Some((_, err)) = &self.rack_init_error {
+            return Err(err.clone());
+        }
+        let State::Uninitialized = self.state else {
+            return Err(ApiError::AlreadyInitialized);
+        };
+        let pkgs = create_pkgs(rack_uuid, initial_membership.clone())
+            .map_err(|e| ApiError::RackInitFailed(e))?;
+        let mut iter = pkgs.expose_secret().into_iter();
+        let our_pkg = iter.next().unwrap().clone();
+
+        // Move into an initialized state
+        self.state = State::InitialMember {
+            pkg: our_pkg,
+            distributed_shares: BTreeMap::new(),
+        };
+
+        let packages: BTreeMap<Baseboard, SharePkg> = initial_membership
+            .into_iter()
+            .filter(|peer| *peer != self.id)
+            .zip(iter.cloned())
+            .collect();
+
+        let _ = self.request_manager.new_init_rack_req(
+            now,
+            rack_uuid,
+            packages.clone(),
+            &self.connected_peers,
+        );
+
+        Ok(())
+    }
+
+    /// Are we still waiting for `InitAck` responses from peers?
+    pub fn is_rack_initializing(&self) -> bool {
+        self.request_manager.has_init_rack_req()
+    }
+
+    /// Initialize a node added after rack initialization
+    ///
+    /// Persistence is required after a successful call to `init_learner`
+    pub fn init_learner(&mut self, now: Instant) -> Result<(), ApiError> {
+        if let Some((_, err)) = &self.rack_init_error {
+            return Err(err.clone());
+        }
+        let State::Uninitialized = self.state else {
+            return Err(ApiError::AlreadyInitialized);
+        };
+
+        if let Some(to) = self.connected_peers.first() {
+            let _ = self.request_manager.new_learn_sent_req(now, to.clone());
```

Comment:
Nit:
```suggestion
            let _uuid = self.request_manager.new_learn_sent_req(now, to.clone());
```
(otherwise, I wonder if you're discarding an error here)

Reply:
good call!